crates/test_utils/wasm/wasm_workspace/validation_package_success/src/lib.rs

```diff
@@ -1,6 +1,57 @@
 use hdk3::prelude::*;
 
+const NUM_SONGS: usize = 30;
+
+#[hdk_entry(id = "song", required_validation_type = "element")]
+struct Song;
+
+#[hdk_entry(id = "artist", required_validation_type = "custom")]
+struct Artist;
+
+entry_defs![Song::entry_def(), Artist::entry_def()];
+
+#[hdk_extern]
+fn validation_package(
+    app_entry_type: AppEntryType,
+) -> ExternResult<ValidationPackageCallbackResult> {
+    let index = app_entry_type.id();
+    match u8::from(index) {
+        // Artist
+        1 => {
+            let query = QueryFilter::new()
+                .entry_type(EntryType::App(AppEntryType::new(
+                    0.into(),
+                    0.into(),
+                    EntryVisibility::Public,
+                )))
+                .include_entries(true);
+            let songs = query!(query)?.0;
+            // Need to post at least 30 songs to be an artist on this dht
+            if songs.len() >= NUM_SONGS {
+                Ok(ValidationPackageCallbackResult::Success(
+                    ValidationPackage::new(songs),
+                ))
+            } else {
+                Ok(ValidationPackageCallbackResult::Fail(
+                    "Need at least 30 songs to be an artist on this dht".to_string(),
+                ))
+            }
+        }
+        _ => Ok(ValidationPackageCallbackResult::Success(
+            ValidationPackage::new(vec![]),
+        )),
```

Comment:
Isn't this `unreachable!()`? Might be clearer to say so, if so.

Reply:
No unfortunately it's used in another test that simply calls this function to prove that it's callable. It calls with an entry_id of 0 though so it works. Not great but I didn't want to create a whole new zome to test this